browser/ui/webui/brave_shields/cookie_consent_page_handler.cc

```diff
@@ -0,0 +1,55 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_shields/cookie_consent_page_handler.h"
+
+#include <utility>
+
+#include "brave/browser/brave_browser_process.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/common/brave_shield_constants.h"
+#include "brave/components/brave_shields/common/pref_names.h"
+#include "chrome/browser/browser_process.h"
+#include "components/prefs/pref_service.h"
+
+CookieConsentPageHandler::CookieConsentPageHandler(
+    mojo::PendingReceiver<brave_shields::mojom::CookieConsentPageHandler>
+        receiver,
+    base::WeakPtr<ui::MojoBubbleWebUIController::Embedder> embedder)
+    : receiver_(this, std::move(receiver)), embedder_(embedder) {
+  DCHECK(embedder_);
+}
+
+CookieConsentPageHandler::~CookieConsentPageHandler() = default;
+
+void CookieConsentPageHandler::ShowUI() {
+  if (embedder_) {
+    embedder_->ShowUI();
+  }
+}
+
+void CookieConsentPageHandler::CloseUI() {
+  if (embedder_) {
+    embedder_->CloseUI();
+  }
+}
+
+void CookieConsentPageHandler::EnableCookieConsentBlocking() {
+  SaveUserInteraction();
+
+  g_brave_browser_process->ad_block_service()
```

Comment:
These are going to be a bit problematic because there isn't a great way to inject these dependencies. I'm thinking `AdBlockRegionalServiceManager::Set` and `AdBlockRegionalServiceManager::Get` similar to `ExtensionsBrowserClient::Set`
wdyt @antonok-edm ?